### ISSUE TITLE
Weird roadrunner/reset issue

### ISSUE THREAD

**Issue by kirichoi** [sys-bio/roadrunner#722]
I have encountered a very odd issue when accessing the output of libroadrunner. Let's say I have a very simple model and I want to simulate it twice with different settings:
```
import tellurium as te
import roadrunner

r = te.loada("""
J0: $Xi -> S1; k0*Xi;
J1: S1 -> $Xo; k1*S1;

Xi = 10.0; Xo = 10.0;
k0 = 0.5; k1 = 0.5;
S1 = 0; 
""")

m1 = r.simulate(0, 20, 11)
r.reset()
m2 = r.simulate(0, 20, 6)

print(m1)

print(m2)
```
Everything looks okay until I try to access `m1` through the console from which I get this:

```
            time,         [S1]
 [[ 6.23042e-307, 4.67297e-307],
  [ 1.69121e-306, 9.34609e-307],
  [ 1.33511e-306, 1.33512e-306],
  [ 6.23038e-307, 6.23054e-307],
  [  9.3461e-307, 8.45594e-307],
  [ 9.34601e-307, 8.45606e-307],
  [ 6.23062e-307, 1.78019e-306],
  [ 6.23057e-307, 1.33511e-306],
  [  1.6022e-306, 1.60218e-306],
  [ 1.78019e-306,  6.8981e-307],
  [ 1.11261e-306, 1.42411e-306]]
```
Any thoughts?

**Comment by kirichoi** [comment]
I suspect this is some sort of assignment issue which is understandable but was not present before. For example, this works perfectly fine:
```
import tellurium as te
import roadrunner
import copy

r = te.loada("""
J0: $Xi -> S1; k0*Xi;
J1: S1 -> $Xo; k1*S1;

Xi = 10.0; Xo = 10.0;
k0 = 0.5; k1 = 0.5;
S1 = 0; 
""")

m1 = copy.copy(r.simulate(0, 20, 11))
r.reset()
m2 = copy.copy(r.simulate(0, 20, 6))

print(m1)

print(m2)
```

**Comment by hsauro** [comment]
Are you sure this didn’t happen in previous versions? I suspect not since
I’ve never seen any thing like this.

**Comment by hsauro** [comment]
That’s not good. I’ll have to try it myself.

**Comment by hsauro** [comment]
Could you make it an issue so we don’t forgot. I assume this won’t be
worked on until next year.

**Comment by kirichoi** [comment]
I can confirm that in previous versions of roadrunner, this was not the case. Using libroadrunner 1.5.4 yields expected results when trying to call variable `m1`:
```
    time,    [S1]
 [[    0,       0],
  [    2, 6.32122],
  [    4, 8.64663],
  [    6, 9.50212],
  [    8, 9.81683],
  [   10, 9.93261],
  [   12, 9.97521],
  [   14, 9.99088],
  [   16, 9.99664],
  [   18, 9.99876],
  [   20, 9.99954]]
```
This feels a lot like an example of one of those variable reference oddities in Python.

**Comment by hsauro** [comment]
Agreed, this is a bad bug. We won't make an official announcement until we
can figure out what is wrong.

**Comment by luciansmith** [comment]
This is now fixed with https://github.com/sys-bio/roadrunner/commit/45aa2265e3b1502a27d0523a022635d87c8cfb4a  Thanks for pointing it out--I had seen a problem with the linux python bindings for the rk45 tests, but erroneously thought that was the only problem.

For the record, the bug was that I inadvertently set the 'copy_results' simulation option to 'false', meaning that any time we created a python object, we only *pointed* at the data, instead of giving Python a new copy of it.  This option is apparently there so that people can speed up their simulations, even though it creates inherently unstable Python objects.
